utils/src/main/scala/za/co/absa/enceladus/utils/testUtils/SparkTestBase.scala

```diff
@@ -17,17 +17,69 @@ package za.co.absa.enceladus.utils.testUtils
 
 import org.apache.log4j.{Level, Logger}
 import org.apache.spark.sql.SparkSession
+import org.apache.spark.SparkConf
+import com.typesafe.config.ConfigFactory
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{Path, FileSystem}
+import scala.collection.JavaConversions._
+import java.io.File
 import za.co.absa.enceladus.utils.time.TimeZoneNormalizer
 
-trait SparkTestBase {
+trait SparkTestBase { self =>
   System.setProperty("user.timezone", "UTC");
 
-  implicit val spark = SparkSession.builder().master("local[*]").appName("test")
-  .config("spark.sql.codegen.wholeStage", false)
-  .config("spark.ui.enabled", "false")
-  .config("spark.driver.bindAddress","127.0.0.1")
-  .config("spark.driver.host", "127.0.0.1")
-  .getOrCreate()
+  val config = ConfigFactory.load()
+  val sparkMaster = config.getString("enceladus.utils.testUtils.sparkTestBaseMaster")
+
+  val sparkBuilder = SparkSession.builder()
+    .master(sparkMaster)
+    .appName(s"Enceladus test - ${self.getClass.getName}")
+    .config("spark.ui.enabled", "false")
+
+  implicit val spark = if (sparkMaster == "yarn") {
+    val confDir = config.getString("enceladus.utils.testUtils.hadoop.conf.dir")
+    val distJarsDir = config.getString("enceladus.utils.testUtils.spark.distJars.dir")
+    val sparkHomeDir = config.getString("enceladus.utils.testUtils.spark.home.dir")
+
+    //load all hadoop configs
+    val hadoopConf = new Configuration()
+    hadoopConf.addResource(new Path(s"$confDir/hdfs-site.xml"))
+    hadoopConf.addResource(new Path(s"$confDir/yarn-site.xml"))
+    hadoopConf.addResource(new Path(s"$confDir/core-site.xml"))
+
+    val hadoopConfigs = hadoopConf.iterator().map(entry => (s"spark.hadoop.${entry.getKey}", entry.getValue)).toMap
+
+    //load spark configs from SPARK_HOME
+    val sparkConfigIn = ConfigFactory.empty().atPath(s"$sparkHomeDir/conf/spark-defaults.conf")
+    val sparkConfigs = sparkConfigIn.entrySet().filter(_.getKey != "spark.yarn.jars").map(entry => (entry.getKey, entry.getValue.unwrapped().toString())).toMap
+
+    val allConfigs = hadoopConfigs ++ sparkConfigs
+
+    //get a list of all dist jars
+    val distJars = FileSystem.get(hadoopConf).listStatus(new Path(distJarsDir)).map(_.getPath)
+
+    //get a list of jars used by the given test - spark jars are taken from dist jars, so include absa deps only
+    val cl = this.getClass.getClassLoader
+    val localJars = cl.asInstanceOf[java.net.URLClassLoader].getURLs.filter(c => c.toString().contains("absa")).map(_.toString())
```

Comment:
If I understand correctly, wouldn't it be better to have here `enceladus` instead of `absa`

Reply:
I guess `absa` for `spline` and such?

Reply:
That would make sense. Is that the reason, @lokm01? Can I close this?

Reply:
Yeah exactly... 
Spark jars (incl hadoop dependencies) would be located in the dist jars on HDFS. Then we need jars that we depend on (but not everything - no need for scala and others)